gateway/test/src/test/java/io/apiman/gateway/test/policies/LoggingPolicy.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2016 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.apiman.gateway.test.policies;
+
+import io.apiman.gateway.engine.beans.ApiRequest;
+import io.apiman.gateway.engine.beans.ApiResponse;
+import io.apiman.gateway.engine.policy.IPolicy;
+import io.apiman.gateway.engine.policy.IPolicyChain;
+import io.apiman.gateway.engine.policy.IPolicyContext;
+import io.apiman.gateway.test.logging.TestLogger;
+
+/**
+ * A simple policy used for testing.
+ *
+ * @author Marc Savy {@literal <msavy@redhat.com>}
+ */
+public class LoggingPolicy implements IPolicy {
+
+    public static void reset() {
+    }
+
+    /**
+     * Constructor.
+     */
+    public LoggingPolicy() {
+    }
+
+    /**
+     * @see io.apiman.gateway.engine.policy.IPolicy#parseConfiguration(java.lang.String)
+     */
+    @Override
+    public Object parseConfiguration(String jsonConfiguration) {
+        return new Object();
+    }
+
+    /**
+     * @see io.apiman.gateway.engine.policy.IPolicy#apply(io.apiman.gateway.engine.beans.ApiRequest, io.apiman.gateway.engine.policy.IPolicyContext, java.lang.Object, io.apiman.gateway.engine.policy.IPolicyChain)
+     */
+    @Override
+    @SuppressWarnings("nls")
+    public void apply(final ApiRequest request, final IPolicyContext context, final Object config,
+            final IPolicyChain<ApiRequest> chain) {
+        // This is just for testing, don't do this at home, kids.
+        TestLogger logger = (TestLogger) context.getLogger(getClass());
+        logger.setHeaders(request.getHeaders());
```

Comment:
TestLogger writes to the headers instead of stdout (or whatever), so it can be tested against using resttests